Source/RunActivity/Viewer3D/Viewer.cs

```diff
@@ -987,28 +991,24 @@ void HandleUserInput(ElapsedTime elapsedTime)
 
             if (UserInput.IsPressed(UserCommand.CameraCab))
             {
-                if (CabCamera.IsAvailable)
+                if (CabCamera.IsAvailable || ThreeDimCabCamera.IsAvailable)
                 {
                     new UseCabCameraCommand(Log);
                 }
-                else if (ThreeDimCabCamera.IsAvailable)
-                {
-                    new Use3DCabCameraCommand(Log);
-                }
                 else
                 {
                     Simulator.Confirmer.Warning(Viewer.Catalog.GetString("Cab view not available"));
                 }
             }
-            if (UserInput.IsPressed(UserCommand.CameraThreeDimensionalCab))
+            if (UserInput.IsPressed(UserCommand.CameraToggleThreeDimensionalCab))
             {
-                if (ThreeDimCabCamera.IsAvailable)
+                if (CabCamera.IsAvailable && ThreeDimCabCamera.IsAvailable)
                 {
-                    new Use3DCabCameraCommand(Log);
+                    new ToggleThreeDimensionalCabCameraCommand(Log);
                 }
                 else
                 {
-                    Simulator.Confirmer.Warning(Viewer.Catalog.GetString("3D Cab view not available"));
+                    Simulator.Confirmer.Warning(Viewer.Catalog.GetString("This car doesn't have both a 2D cab and a 3D cab"));
```

Comment:
```suggestion
                    Simulator.Confirmer.Warning(Viewer.Catalog.GetString("Cab view not available"));
```
This grammar is incorrect, and a better version would be "This car has neither a 2D nor a 3D cab". However, perhaps "Cab view not available" also works here, and we already have translations for that string.

Reply:
Nope.
Here, I want to say that there isn't a 2D and a 3D at the same time.
You can't change from 2D to 3D if both are not there.

Reply:
Very well; let's go with "This car has neither a 2D nor a 3D cab". I have updated the suggestion.

Reply:
"neither nor" means both are not there.
Here, one can be present but not the other.

Reply:
I see. Language can be difficult, so here's a truth table:
```
                                         +--+--+
                                         |2D|3D|
+----------------------------------------+--+--+
|"This car does not have a 2D cab"       |N |Y |
+----------------------------------------+--+--+
|"This car does not have a 3D cab"       |Y |N |
+----------------------------------------+--+--+
|"This car does not have another type    |N |Y |
| of cab"                                |Y |N |
+----------------------------------------+--+--+
|"This car has neither a 2D nor a 3D cab"|N |N |
+----------------------------------------+--+--+
```
I still suggest "This car does not have another type of cab," because when I read "This car doesn't have both a 2D cab and a 3D cab," my mind instantly jumps to the N/N case.

Or, preferably, we test for either a missing 2D or 3D cab and display one of the top two strings.

Reply:
For me, my proposal means not(Y/Y).

Between your proposals, I prefer to test and to display which one is missing.